src/mod/endpoints/mod_sofia/sofia_glue.c

```diff
@@ -3294,6 +3294,65 @@ void sofia_glue_build_vid_refresh_message(switch_core_session_t *session, const
 }
 
 
+char *sofia_glue_get_encoded_fs_path(nua_handle_t *nh, sip_route_t *rt, switch_bool_t add_fs_path_prefix)
+{
+	char *route;
+	int count = 0;
+	char *routes[ROUTE_MAX_HEADERS] = {0};
```

Comment:
Do we have to limit the amount of routes?

Reply:
There is a max forward which will trigger for a message exceeding some hops (70 is standard) so this would be more than enough. And I haven't seen in practice more than 6 routes ever.

Reply:
Many sip implementations have a limit of 6 Max

https://www.dialogic.com/webhelp/BorderNet2020/2.2.0/WebHelp/sip_message_header_restrict.htm

Reply:
I meant to use ROUTE_MAX_HEADERS = 10 later in the code.

Reply:
I agreed, 6 `Route` headers are a small value.
In out use case we already use 6 `Route` headers example
```
Record-Route:  <sip:[2600:1f14:6d8:5408::100];r2=on;lr=on;ftag=7QXK8e535USSQ;did=971.57e2>
Record-Route:  <sip:[2600:1f14:6d8:5408::100]:5080;transport=tcp;r2=on;lr=on;ftag=7QXK8e535USSQ;did=971.57e2>
Record-Route:  <sip:[2600:1f1c:6d5:aa01::7];transport=tcp;r2=on;lr=on;did=971.aa9>
Record-Route:  <sip:3.101.84.135;transport=tcp;r2=on;lr=on;did=971.aa9>
Record-Route:  <sip:192.76.120.10;transport=tcp;r2=on;lr;ftag=7QXK8e535USSQ>
Record-Route:  <sip:10.255.0.1;r2=on;lr;ftag=7QXK8e535USSQ>
```
Here is
1. 10.255.0.1 - Telnyx internal proxy;
2. 192.76.120.10 - Telnyx SBC;
3. 3.101.84.135 - IPv4 interface of our SBC;
4. 2600:1f1c:6d5:aa01::7 - IPv6 interface of our SBC;
5. 2600:1f14:6d8:5408::100, TCP transport and port 5080 - the interface of our internal proxy with disabled NAT feature (on port 5060 force enabled NAT features for messages from the Internet);
6. 2600:1f14:6d8:5408::100, TCP transport and port 5060 - the interface of our internal proxy that interacts with FreeSwitch.

So when using IPv6 networks and using multiple types of transport protocols (UDP/TCP/TLS/WSS) then we can easily reach 6 `Route` headers.
10 `Route` headers value looks safer, for me.